adafruit_slideshow.py

```diff
@@ -0,0 +1,209 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""
+`adafruit_slideshow`
+====================================================
+CircuitPython helper library for displaying a slideshow of images on a board with a built-in
+display.
+
+* Author(s): Kattni Rembor, Carter Nelson
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+ * `Adafruit Hollowing M0 Express <https://www.adafruit.com/product/3900>`_"
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+import time
+import os
+import random
+import board
+import displayio
+import pulseio
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Slideshow.git"
+
+
+class SlideShow():
+    """Class for displaying a slideshow of .bmp images on boards with built-in
+    displays.
+
+    :param str folder: Specify the folder containing the image files, in quotes. Default is
+                       the root directory, ``"/"``.
+
+    :param order: The order in which the images display. You can choose random (``RANDOM``) or
+                  alphabetical (``ALPHA``). Default is ``RANDOM``.
+
+    :param loop: Specify whether to loop the images or play through the list once. `True`
+                 if slideshow will continue to loop, ``False`` if it will play only once.
+                 Default is ``True``.
+
+    :param int dwell: The number of seconds each image displays, in seconds. Default is 3.
+
+    Example code for Hollowing Express. With this example, the slideshow will play through once
+    in alphabetical order:
+
+    .. code-block:: python
+
+        import adafruit_slideshow
+
+        slideshow = adafruit_slideshow.SlideShow()
+        slideshow.loop = False
+        slideshow.order = slideshow.ALPHA
+
+        while slideshow.update():
+            pass
+    """
+    MAX_BRIGHTNESS = 2 ** 15
+
+    ALPHA = 0
+    RANDOM = 1
+
+    FADE_IN = 0
+    SHOW_IMG = 1
+    FADE_OUT = 2
+    LOAD_IMG = 3
```

Comment:
These are all internal variables. Should start with `_`? (`_FADE_IN`)

Also, now that I read further down...these should also move outside of the `SlideShow()` class. The variables themselves don't change, the `self._current_state` does, so you don't need to recreate them for every instance of the class. Calls to `self.FADE_IN` etc will also need to change after that.